File: leetcode/345.反转字符串中的元音字母.java
import java.util.Arrays;

/*
 * @lc app=leetcode.cn id=345 lang=java
 *
 * [345] 反转字符串中的元音字母
 */

// @lc code=start
class Solution {

    public String reverseVowels(String s) {
        int l = 0, r = s.length() - 1;
        StringBuilder sb = new StringBuilder(s);
        Character lowerChar[] = { 'a', 'e', 'i', 'o', 'u' };
        Character upperChar[] = { 'A', 'E', 'I', 'O', 'U' };
        while (l < r) {
            char rChar = s.charAt(r), lChar = s.charAt(l);
            if (!Arrays.asList(lowerChar).contains(rChar) && !Arrays.asList(upperChar).contains(rChar)) {
                r--;
            } else if (!Arrays.asList(lowerChar).contains(lChar)
                    && !Arrays.asList(upperChar).contains(lChar)) {
                l++;
            } else {
                char temp = lChar;
                sb.setCharAt(l, rChar);
                sb.setCharAt(r, temp);
                r--;
                l++;
            }
        }
        return sb.toString();
    }

}
// @lc code=end

File: 345.反转字符串中的元音字母.py
#
# @lc app=leetcode.cn id=345 lang=python3
#
# [345] 反转字符串中的元音字母
#

# @lc code=start
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        target = "aeiouAEIOU"
        l, r = 0, len(s) - 1
        while l <= r:
            if s[l] not in target:
                l += 1
                continue
            if s[r] not in target:
                r -= 1
                continue
            temp = s[l]
            s[l] = s[r]
            s[r] = temp
            l += 1
            r -= 1
        return ''.join(s)
# s = Solution()
# s.reverseVowels("hello")
            
# @lc code=end

